### PR TITLE
feat: adding nft-fee-transaction example

### PR DESCRIPTION
**Pull request by marcus-girolneto**
### Changes
+ Adding the NFT example with FeeTransactions support
+ Updated README
+ Added JavaDocs to the newer lifecycle functions

### Notes
+ We still need to update the tessellation version to match the version released with FeeTransactions

### CODE REVIEW

File: examples/nft-fee-transaction/gitignore
Comment (scasplte2) [inline examples/nft-fee-transaction/gitignore:1]: duplicate file of the `.gitignore`?

File: examples/nft-fee-transaction/project/Dependencies.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import sbt.*
+
+object Dependencies {
+
+  object V {
+    val tessellation = "99.99.99" //TODO: Update with the official version with FeeTransactions
```
Comment (scasplte2) [inline examples/nft-fee-transaction/project/Dependencies.scala:6]: do you want to update this to a tagged version?

**Review by scasplte2 — APPROVED** [review]

File: examples/nft-fee-transaction/project/Dependencies.scala (hunk 1 above)
Comment (marcus-girolneto) [reply examples/nft-fee-transaction/project/Dependencies.scala:6]: We don't have a fixed version in testnet, this should be updated once we migrate to intnet

Comment (marcinwadon) [reply examples/nft-fee-transaction/project/Dependencies.scala:6]: I think it would be `3.0.0`. but if we can, let's wait for release

**Pull request merged**